tests/integration/test_healing.py

```diff
@@ -293,3 +278,139 @@ async def test_replicate_data_on_restart(ops_test: OpsTest):
                 assert random_chars in output, "❌ Data was not synced"
     except RetryError:
         assert False, "❌ Data was not synced"
+
+
+@pytest.mark.order(5)
+@pytest.mark.abort_on_fail
+@pytest.mark.healing_tests
+async def test_cluster_pause(ops_test: OpsTest):
+    """Pause test.
+
+    A graceful simultaneous restart of all instances,
+    check primary election after the start, write and read data
+    """
+    app = await app_name(ops_test)
+    all_units = ops_test.model.applications[app].units
+
+    config = {
+        "username": CLUSTER_ADMIN_USERNAME,
+        "password": await get_system_user_password(all_units[0], CLUSTER_ADMIN_USERNAME),
+    }
+
+    # stop all instances
+    logger.info("Stopping all instances")
+    for unit in all_units:
+        await graceful_stop_server(ops_test, unit.name)
+
+    # verify connection is not possible to any instance
+    for unit in all_units:
+        unit_ip = await get_unit_ip(ops_test, unit.name)
+        config["host"] = unit_ip
+        assert not is_connection_possible(
+            config
+        ), f"❌ connection to unit {unit.name} is still possible"
+
+    # restart all instances
+    logger.info("Starting all instances")
+    for unit in all_units:
+        await start_server(ops_test, unit.name)
+
+    async with ops_test.fast_forward():
+        # trigger update status asap
+        # which in has self healing handler
+
+        # verify all instances are accessible
+        for unit in all_units:
+            unit_ip = await get_unit_ip(ops_test, unit.name)
+            config["host"] = unit_ip
+            assert is_connection_possible(
```

Comment:
Do you still need the `fast_forward` context manager after this point? Surely if connections are possible, self-healing has been carried out?

Reply:
Not strictly but it will greatly speed up the test. because the treatment is being done on the `update-status` event handler.
The `is_connection_possible` will assert that the mysql-server instance is up and running, but not necessarily the instance within the cluster. This is only fully validated after the write-n-read tests that follows.